### PR TITLE
Adding generated docs

### PR DESCRIPTION
**Pull request by qureshi-ali**

### CODE REVIEW

**Comment by melonpii** [comment]
Doc generation has to be automated like the sample hw repo.

**Comment by qureshi-ali** [comment]
Yep working on that right now

**Review by atharv67 — APPROVED** [review]

File: .github/workflows/sphinx-autogenerator.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        steps:
+            - uses: actions/checkout@v2
+            - name: Build documentation and commit
+              run: |
+                  git pull
+                  cd docs/
+                  sudo apt-get install python3-sphinx
+                  sudo rm code.rst
+                  sudo rm modules.rst
+                  sphinx-apidoc -o . ../code/
```
Comment (melonpii) [inline .github/workflows/sphinx-autogenerator.yml]: sphinx-apidoc -o . ../src/

File: .github/workflows/sphinx-autogenerator.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    # This workflow contains a single job called "build"
+    build:
+        # The type of runner that the job will run on
+        runs-on: ubuntu-latest
+
+        steps:
+            - uses: actions/checkout@v2
+            - name: Build documentation and commit
+              run: |
+                  git pull
```
Comment (melonpii) [inline .github/workflows/sphinx-autogenerator.yml]: > git pull origin main.

We are not on any branch. So this will fail.

Comment (melonpii) [reply .github/workflows/sphinx-autogenerator.yml]: Try adding git config pull.rebase false

**Comment by qureshi-ali** [comment]
Closing branch without accepting pr

**Comment by qureshi-ali** [comment]
Closing PR

**Comment by qureshi-ali** [comment]
.

**Pull request closed without merging**